### PR TITLE
Fix typings

### PR DESCRIPTION
**Pull request by evelynhathaway**
Rebased from @Wicusiu's PR #128

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -149,7 +151,7 @@ declare module 'ical-generator' {
       events(events: EventData[]): ICalCalendar;
       save(path: string, cb: Function): ICalCalendar;
       saveSync(path: string): number;
-      serve(respone: Response, filename: string): ICalCalendar;
+      serve(respone: http.ServerResponse, filename: string): ICalCalendar;
```
Comment (evelynhathaway) [inline index.d.ts]: Solves TS error:
```
/index.d.ts:152:22 - error TS2304: Cannot find name 'Response'.
152       serve(respone: Response, filename: string): ICalCalendar;
```

File: index.d.ts [hunk 2]
```diff
@@ -219,7 +221,7 @@ declare module 'ical-generator' {
       type(): attendeeType;
       type(type: attendeeType): ICalAttendee;
       delegatedTo(): string | ICalAttendee;
-      delegatedTo(attendee: string | ICalAttendee);
+      delegatedTo(attendee: string | ICalAttendee): ICalAttendee;
```
Comment (evelynhathaway) [inline index.d.ts]: @Wicusiu added a return type, and I aligned it to match the other overloads. See: https://github.com/sebbo2002/ical-generator/pull/128#pullrequestreview-182538188

**Comment by sebbo2002** [comment]
Great work, thank you! ❤️

**Pull request merged**